api/controllers/user.js

```diff
@@ -36,17 +36,28 @@ module.exports = {
       return;
     }
 
-    // save new user
-    user = new User({
-      username: req.body.username,
-      password: req.body.password
-    });
+    const { username } = req.body;
+
+    // check if username is already in db
+    User.findOne({ username }).then(user => {
```

Comment:
One thing going on here is that the save is reusing the "user" variable returned from the findOne, which will work but probably best being called something else to be clearer. 

That said probably don't need to worry about that as this could do with modifying to use promises/async/await to avoid the save being nested inside inside findOne, as once this code begins expanding it could get messy